mlflow/store/abstract_store.py

```diff
@@ -197,7 +198,7 @@ def get_metric_history(self, run_uuid, metric_key):
         pass
 
     @abstractmethod
-    def search_runs(self, experiment_ids, search_filter, run_view_type):
+    def search_runs(self, experiment_ids, search_filter, run_view_type, max_results):
```

Comment:
Can we make this an optional argument (with default of `SEARCH_MAX_RESULTS_DEFAULT`) to avoid breaking the AbstractStore search_runs API?